web/components/contract/like-market-button.tsx

```diff
@@ -35,8 +35,20 @@ export function LikeMarketButton(props: {
     if (!user) return firebaseLogin()
 
     setIsLiking(true)
-    likeContract(user, contract).catch(() => setIsLiking(false))
-    toast(`You tipped ${contract.creatorName} ${formatMoney(LIKE_TIP_AMOUNT)}!`)
+    const timeoutId = setTimeout(() => {
+      likeContract(user, contract).catch(() => setIsLiking(false))
+    }, 3000)
```

Comment:
Did you mean to put TIP_UNDO_DURATION instead of 3000?

Reply:
no, I figured timeout should be slightly longer than when the toast is on screen to avoid a race condition, but I'll make it not a magic number lol